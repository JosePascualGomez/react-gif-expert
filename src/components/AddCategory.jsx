import { useState } from 'react'
import { PropTypes } from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('')

    //funcion que se ejecuta cuando estoy digitando en el input y al dar enter ejecutar algo
    const onInputChange = (event) => {
        //captura el dato del input y lo agrega al hook
        setInputValue(event.target.value);
    }

    //Se ejecuta cuando de enter del input
    const onSubmit = (event) => {
        //evitar que se refresque el navegador
        event.preventDefault();
        const valueReturn =inputValue.trim();
        if (valueReturn.length < 1) return;
        onNewCategory(valueReturn);
        setInputValue('');
    }

    return (
        //aria.label necesario para referenciarlo en la prueba por getByRole
        <form aria-label='form' onSubmit={(event) => onSubmit(event)}>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={(event) => onInputChange(event)}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}