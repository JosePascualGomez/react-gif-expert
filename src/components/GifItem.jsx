

export const GifItem = ({ title, url, id}) => {
    console.log(title, url);
    return (
        <div class="card">
            <img src={url} alt={id} />
            <p>{title}</p>
        </div>
    )
}
