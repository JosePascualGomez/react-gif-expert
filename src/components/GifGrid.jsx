
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";


export const GifGrid = ({ category }) => {

    // useEffect(() => {
    //   getGifs(category)
    //   .then(newImages => setImages(newImages));
    // }, [])
    
    //Se realiza a traves de un hook con el fin de que no se ejecute cada vez que se renderice
    const {images, isLoading} = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            
            
            {
                isLoading 
                ?(<h2>Cargando...</h2>)
                :null
            }

{
                isLoading && (<h2>Cargando...</h2>)
            }
            

            <div className="card-grid">
                {
                    images.map((img) => (
                        <GifItem key={img.id}
                            //esparce las properties del img con spread
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
