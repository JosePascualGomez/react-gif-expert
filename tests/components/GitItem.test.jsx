
import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

const title = 'MiTitulo';
const url ='https://pagina/foto.jpg'


describe('Pruebas del Git Item', () => { 
    test('Hacer match con SnapShop', () => { 
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
     });

     test('mostrar la imagen con el URL y el ALT indicado', () => { 
        render(<GifItem title={title} url={url} />);
        //expect(screen.getByRole('img').src).toBe(url);
        //expect(screen.getByRole('img').alt).toBe(title);    
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
      })

      test('Debe mostrar el titulo en el componente', () => { 
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
      })

 })
