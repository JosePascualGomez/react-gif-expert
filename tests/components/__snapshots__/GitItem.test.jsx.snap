// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del Git Item Hacer match con SnapShop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="MiTitulo"
      src="https://pagina/foto.jpg"
    />
    <p>
      MiTitulo
    </p>
  </div>
</div>
`;
